app/server/models/ussd.py

```diff
@@ -0,0 +1,32 @@
+from sqlalchemy.dialects.postgresql import JSON
+
+from server import db
+from server.models.utils import ModelBase
+
+class UssdMenu(ModelBase):
+    __tablename__ = 'ussd_menus'
+
+    name = db.Column(db.String, nullable=False)
+    description = db.Column(db.String)
+    parent_id = db.Column(db.Integer)
+    """
+        TODO: change how we do i18n
+        keeping this as is from migration from grassroots economics
+        but this system is really bad since we always need 
+        all languages defined for all new menu items, no graceful fallback 
+    """
+    display_text_en = db.Column(db.String, nullable=False)
+    display_text_sw = db.Column(db.String, nullable=False)
+
+class UssdSession(ModelBase):
+    __tablename__ = 'ussd_sessions'
+
+    session_id = db.Column(db.String, nullable=False, index=True, unique=True)
```

Comment:
i only saw one other index in all of the codebase. is this correct? that's a bit worrying for performance

Reply:
good catch. that's probably just me being shit. i'm never really sure when to index and when not to, which leads to a lack of indexing in general.

Reply:
a super quick primer on indices:
- when you do a db query on a field (`User.query.filter_by(phone=msisdn)` for example), it scans all your rows looking at that entry, meaning O(n) search. when you have millions of rows, that's real bad
- when you add an index, it essentially creates another whole copy of your table, but sorted by whatever you indexed by. this means 2x (roughly, in hand wavy terms) storage costs and slightly slower write time, like plus O(ln n)? more hand wavy tree insertion algorithms, but querying by that field becomes essentially O(1)
- when you do a sort by, find by, basically any sort of sql operation, sql magically knows how to use the appropriate indices for that action (asterisk here, it sometimes messes up if you have too many similar indices on the same table. then you gotta go hunt down what's causing it to use the wrong one and that's sad times)
- if you're doing a search on multiple fields at once, you'll want a multi-field index to account for that (`db.Index(<name>, <field1>, <field2>, etc, <params>)`)

summary is: if you're often searching, sorting, etc, by a specific field or combination of fields, you should index on that to increase performance. storage cost is cheap these days so over-indexing is generally better than under-indexing

Reply:
oh man that's _super_ helpful. So we should be indexing any field that get's a relatively commonly used filter_by query?

Reply:
yep! filter_by, order_by, etc. i'm not sure if id needs to be indexed, it was done automagically in rails so idk if that was rails or postgres that did it

Reply:
it's not as big a deal for tables that stay small, but for tables that'll get big real fast like CreditTransfer, it's important to have indices